### ISSUE TITLE
fs.readdirSync is not working properly when compiled

### ISSUE THREAD

**Issue by achekulaev** [zeit/pkg#813]
**1. Create main.js**

```javascript
console.log(process.version);
console.log(
    require('fs').readdirSync('.', { withFileTypes: true })
);
```

**2. Execute via node:** `node main.js`

```
§ node main.js 
v12.13.1
[
  Dirent { name: 'main', [Symbol(type)]: 1 },
  Dirent { name: 'main.js', [Symbol(type)]: 1 },
]
```

**3. Package with** ` pkg main.js -t macos` (pkg@4.4.2)

**4. Execute binary**

```
§ ./main 
v12.13.1
[ 'main', 'main.js', 'snapshot' ]
```

**Expected result**

1. Because `withFileTypes` is passed, the result of the function should be an array or `fs.Dirent` entities, like it is when executed via `node main.js`, not an array of Strings, as it appeared when executed from the packaged binary.

2. (nice to have) There should be no `snapshot` folder in the readdir results

**Comments**

I am not really sure how this is possible or how to work around the issue, since it is the [core NodeJS function](https://nodejs.org/api/fs.html#fs_fs_readdirsync_path_options) not working properly when compiled into binary.

This error leads to [nodelib](https://github.com/nodelib/nodelib) not working properly, [fast-glob](https://github.com/mrmlnc/fast-glob) that is dependent on it not working properly, and [oclif](https://github.com/oclif/oclif) that is dependent on fast-glob not working properly when compiled into binary with zeit/pkg.

Errors look like this (putting it here for others to be able to google them):

```
TypeError: entry.dirent.isSymbolicLink is not a function
    at /snapshot/doo/node_modules/@nodelib/fs.scandir/out/providers/sync.js:23:26
    at Array.map (<anonymous>)
    at readdirWithFileTypes (/snapshot/doo/node_modules/@nodelib/fs.scandir/out/providers/sync.js:17:20)
    at Object.read (/snapshot/doo/node_modules/@nodelib/fs.scandir/out/providers/sync.js:8:16)
    at SyncReader.scandirSync [as _scandir] (/snapshot/doo/node_modules/@nodelib/fs.scandir/out/index.js:16:17)
    at SyncReader._handleDirectory (/snapshot/doo/node_modules/@nodelib/fs.walk/out/readers/sync.js:28:34)
    at SyncReader._handleQueue (/snapshot/doo/node_modules/@nodelib/fs.walk/out/readers/sync.js:23:18)
    at SyncReader.read (/snapshot/doo/node_modules/@nodelib/fs.walk/out/readers/sync.js:15:14)
    at SyncProvider.read (/snapshot/doo/node_modules/@nodelib/fs.walk/out/providers/sync.js:11:29)
    at ReaderSync.walkSync [as _walkSync] (/snapshot/doo/node_modules/@nodelib/fs.walk/out/index.js:18:21)
```

**Comment by guillaumek** [comment]
found it on google :)
did you manage to get a workaround ?
Best.

**Comment by achekulaev** [comment]
@guillaumek well yes, workaround is do not use `withFileTypes` and write a custom loop instead.

**Comment by jooshkins** [comment]
Ditto, I am having the same problem when trying to use pkg with [oclif](https://github.com/oclif).

**Comment by jooshkins** [comment]
I started trying to fix pkg, and got it somewhat working, but not all the way so I gave up....
 
Anyway, not ideal, but if anyone is looking for a workaround:

**Patch @nodelib to work with pkg**

1. Edit async provider `/node_modules/@nodelib/fs.scandir/out/providers/async.js`
- Add under line 18:
```
const convertToDirent = (list) => {
  const fs = require('fs');
  return list.map(file => new fs.Dirent(file, 1))
}
dirents = convertToDirent(dirents);
```

2. Edit sync provider `/node_modules/@nodelib/fs.scandir/out/providers/sync.js`
- Replace line 14:
` const dirents = settings.fs.readdirSync(directory, { withFileTypes: true });`

**With:**
```
const convertToDirent = (list) => {
  const fs = require('fs');
  return list.map(file => new fs.Dirent(file, 1))
}
 const dirents = convertToDirent(settings.fs.readdirSync(directory));
```

**Comment by igorklopov** [comment]
Please try again with `pkg@4.4.4`

**Comment by achekulaev** [comment]
The problem is solved with 4.4.4. Thank you so much!

**Comment by camow7** [comment]
This issue seems to have back. I'm using `pkg@5.6.0`
